packages/@okta/vuepress-site/docs/guides/oin-oidc-guide/protocol-level-requirements/index.md

```diff
@@ -2,52 +2,98 @@
+
+You need to add code in your SPA app to create the code verifier and code challenge.
+
+The Authorization Code flow with PKCE looks like this for an OIN app:
+
+* Your end user (also known as the Resource Owner) opens the SPA app in their browser so that they can access and work with your application (the Client). The Client requires information from various back-end infrastructure elements (the Resource Server), usually an API or service. By clicking on a sign-in link, the end user grants authorization to your application and initiates the Authorization Code flow with PKCE.
+* Your client application prepares an authentication request to authenticate the end user. This request is composed of specific required parameters and is sent to the Okta `/authorize` endpoint with a code challenge:
+  * `client_id` &mdash; this value is the Okta client ID created for you when you made your Okta app integration and is available in your Okta app integration settings
+  * `redirect_uri` &mdash; the location where the response is sent. This value, which must start with `https://`, is set in your Okta app integrations settings
+  * `response_type=code` &mdash; this informs Okta (as the authorization server) that you want to get back an access token and an ID token in exchange for the authorization code
+  * `state` &mdash; an arbitrary value returned in the token. The client application can use it to remember the state of its interaction with the end user at the time of the authentication call
+  * `scope` &mdash; must contain `openid` to indicate that you want to get back an ID token from the `/token` endpoint
+  * `code_challenge` &mdash; the PKCE code challenge you previously generated
+  * `code_challenge_method` &mdash; the hash method used to generate the challenge value; this value is always `S256`
+* Your application sends a redirect response to the browser containing the composed authentication request. The browser interprets the redirect and sends the authentication request to the `/authorize` endpoint on the Okta authorization server over secure HTTPS.
+* Okta then sends the browser to an Okta sign-in page which asks the end user to provide authentication and consent to the requested scopes, or confirms that the user is already authenticated.
+* The end user provides proof of identity (using any of the supported authentication methods) and gives consent for any requested claims defined in your `scopes` parameter. The claims are specific pieces of information to be included in the token provided by Okta.
+* The browser receives an authorization code response from the Okta authorization server. If there is an error or the authentication fails for any reason, the browser returns the error to the end user. If the authentication is successful, the Okta authorization server issues an authorization code value and includes it in a redirect that sends the browser over to the Client at the **Login Redirect URI** callback location.
```

Comment:
```suggestion
* The browser receives an authorization code response from the Okta Authorization Server. If there is an error or the authentication fails for any reason, the browser returns the error to the end user. If the authentication is successful, the Okta Authorization Server issues an authorization code value and includes it in a redirect that sends the browser over to the client at the **Login Redirect URI** callback location.
```

Reply:
Leaving "the Client" as a proper noun. All other changes made.